### PR TITLE
136024579 create a storage module and angular interceptor

### PR DESCRIPTION
**Pull request by raphaelakpan**
#### What does this PR do?
Stores JWT token in local storage upon successful login / signup and uses an interceptor to store token in header requests. 
#### Description of Task to be completed?
* Create a storage module to store app-wide information such as tokens.
* Create an angular interceptor that attaches tokens to every http request if available
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#136024489
#136024579
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: public/js/services/interceptor.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+angular.module('mean.system')
+  .factory('AuthInterceptor', ['LocalStorage', function(LocalStorage) {
+    var AuthInterceptor = {
```
[the remark this answers is not in the document]
Comment (victor-enogwe) [reply public/js/services/interceptor.js:3]: careful when you are accepting hounds let and var suggestions, let scopes your variable definition and might cause issues with the global this.

File: public/js/services/interceptor.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    request: function(config) {
+      if (LocalStorage.getToken('token')) {
+        config.headers['x-access-token'] = LocalStorage.getToken('token');
+      }
+      return config;
+    }
+  };
+  return AuthInterceptor;
```
[the remark this answers is not in the document]
Comment (andela-oakinseye) [reply public/js/services/interceptor.js:11]: I think this is one of the few ESlint issues we can attend to.

File: app/controllers/users.js [hunk 3]
```diff
@@ -39,21 +39,22 @@ exports.signup = function(req, res) {
  */
 exports.signout = function(req, res) {
   req.logout();
-  res.redirect('/');
+  res.redirect('/#!/removeToken/' + 1);
```
Comment (neddinn) [inline app/controllers/users.js]: There is no need having to explicitly add a 1 here. Just check that a token exists in localstorage and remove it.

File: public/js/services/storage.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+angular.module('mean.system')
+  .factory('LocalStorage', ['localStorageService', function(localStorageService) {
```
Comment (neddinn) [inline public/js/services/storage.js]: Please change the factory name to `Storage`. It abstracts the happenings here and still describes it. This is still useful if we decide to change the storage module.

File: public/js/services/storage.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    return {
+      storeToken: function(key, value) {
```
Comment (neddinn) [inline public/js/services/storage.js]: Please make the methods here more generic. We dont want to store only token information. You can remove the `Token` from the name

**Review by neddinn — APPROVED** [review]

**Pull request merged**